### PR TITLE
Fix `_csv.Dialect.__init__`

### PR DESCRIPTION
**Pull request by sobolevn**
All props can be passed to `__init__` as pos-or-keyword:

```python
>>> _csv.Dialect(1)
<_csv.Dialect object at 0x106604d70>
>>> _csv.Dialect(dialect=1)
<_csv.Dialect object at 0x106605d90>

>>> _csv.Dialect(delimiter='.')
<_csv.Dialect object at 0x106605790>

>>> _csv.Dialect(quotechar='.')
<_csv.Dialect object at 0x106605970>
>>> _csv.Dialect(quotechar=None)
<_csv.Dialect object at 0x106605850>
```

Source: https://github.com/python/cpython/blame/e6264b44dc7221c713b14dfa0f5929b33d362829/Modules/_csv.c#L389

### CODE REVIEW

**Comment by sobolevn** [comment]
Ok, I see: `# actually csv.Dialect is a different class to _csv.Dialect at runtime, but for typing purposes, they're identical`

**Comment by srittau** [comment]
See #3613 for why this was added. `csv.Dialect` looks to be a weird hack at runtime, probably back from a time when C classes couldn't be sub-classed.

**Comment by JelleZijlstra** [comment]
Do you still want to change anything here? CI is failing.

**Comment by sobolevn** [comment]
We want:
- "Base class" `_csv.Dialect` must have `__init__` with parameters
- "Child class" `csv.Dialect` must have `__init__(self)`

I can make `csv.Dialect` a subclass of `_csv.Dialect` and make unsafe `__init__` redefinition with type ignores. Will try that :)

**Comment by sobolevn** [comment]
@JelleZijlstra thanks for the reminder!

File: stdlib/_csv.pyi [hunk 1]
```diff
@@ -29,9 +31,18 @@ class Dialect:
     lineterminator: str
     quoting: _QuotingType
     strict: bool
-    def __init__(self) -> None: ...
-
-_DialectLike: TypeAlias = str | Dialect | type[Dialect]
+    def __init__(
```
Comment (srittau) [inline stdlib/_csv.pyi:34]: Shouldn't this be `__new__`?

File: stdlib/_csv.pyi [hunk 2]
```diff
@@ -29,9 +31,18 @@ class Dialect:
...
+        self,
+        dialect: _DialectLike = ...,
```
Comment (srittau) [inline stdlib/_csv.pyi:36]: It seems that `None` is also a valid type.

File: stdlib/_csv.pyi (hunk 1 above)
Comment (sobolevn) [reply stdlib/_csv.pyi:34]: It is `__new__` for `_csv.Dialect`, but `__init__` for `csv.Dialect`. I feel safer to use `__init__` in both cases.

**Review by srittau — APPROVED** [review]

**Pull request merged**